cpp/src/arrow/compute/key_hash.cc

```diff
@@ -383,19 +383,30 @@ void Hashing32::HashMultiColumn(const std::vector<KeyColumnArray>& cols,
   uint32_t num_rows = static_cast<uint32_t>(cols[0].length());
 
   constexpr uint32_t max_batch_size = util::MiniBatch::kMiniBatchLength;
+  const uint32_t alloc_batch_size = std::min(num_rows, max_batch_size);
+  const int64_t estimate_alloc_size = EstimateBatchStackSize<uint32_t>(alloc_batch_size);
```

Comment:
Can we use `auto` here?

```suggestion
  const auto alloc_batch_size = std::min(num_rows, max_batch_size);
  const auto estimate_alloc_size = EstimateBatchStackSize<uint32_t>(alloc_batch_size);
```

Reply:
done